### PR TITLE
[Security Hardened Kubernetes Cluster] Rule 2007 implementation

### PR DESCRIPTION
**Pull request by georgibaltiev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Part of #356 

**Special notes for your reviewer**:

**Release note**:

```feature user
Implementation for rule `2007` from the `security-hardened-k8s` ruleset for provider `managedk8s`.
```

### CODE REVIEW

**Review by AleksandarSavchev — APPROVED** [review]
/lgtm

File: pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2007.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+	namespaces, err := kubeutils.GetNamespaces(ctx, r.Client)
+	if err != nil {
+		return rule.Result(r, rule.ErroredCheckResult(err.Error(), rule.NewTarget("kind", "namespaceList"))), nil
+	}
+
+	var (
+		checkResults []rule.CheckResult
+		checkRules   = func(policyRules []rbacv1.PolicyRule, accepted bool, justification string, target rule.Target) rule.CheckResult {
+			msg := "RBAC Role is accepted to use \"*\" in policy rule verbs."
```
Comment (dimityrmirchev) [inline pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2007.go:86]:
```suggestion
			msg := "Role is accepted to use \"*\" in policy rule verbs."
```

File: pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2007.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+			}
+
+			for _, policyRule := range policyRules {
+				for _, verb := range policyRule.Verbs {
+					if strings.Contains(verb, "*") {
+						if accepted {
+							return rule.AcceptedCheckResult(msg, target)
+						} else {
+							return rule.FailedCheckResult("RBAC Role uses \"*\" in policy rule verbs.", target)
+						}
```
Comment (dimityrmirchev) [inline pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2007.go:98]:
```suggestion
						} 
							return rule.FailedCheckResult("Role uses \"*\" in policy rule verbs.", target)
						
```

File: pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2007.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+					}
+				}
+			}
+			return rule.PassedCheckResult("RBAC Role does not use \"*\" in policy rule verbs.", target)
```
Comment (dimityrmirchev) [inline pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2007.go:102]:
```suggestion
			return rule.PassedCheckResult("Role does not use \"*\" in policy rule verbs.", target)
```

**Review by dimityrmirchev — CHANGES_REQUESTED** [review]
Just some minor requests

File: example/config/managedk8s.yaml [hunk 4]
```diff
@@ -210,6 +210,34 @@ providers:                   # contains information about known providers
...
+    #         foo: bar
+    #       namespaceMatchLabels:
+    #         foo: bar
+    #       justification: "justification"
+    #     acceptedClusterRoles:
+    #     - matchLabels:
+    #         foo: bar
+    #       namespaceMatchLabels:
+    #         foo: bar
+    #       justification: "justification"
```
Comment (AleksandarSavchev) [inline example/config/managedk8s.yaml:226]:
```suggestion
    #         foo: bar
    #       justification: "justification"
```

File: example/config/managedk8s.yaml [hunk 5]
```diff
@@ -210,6 +210,34 @@ providers:                   # contains information about known providers
...
```
Comment (AleksandarSavchev) [inline example/config/managedk8s.yaml:240]:
```suggestion
    #         foo: bar
    #       justification: "justification"
```

**Review by AleksandarSavchev — CHANGES_REQUESTED** [review]
ClusterRoles are not namespaces

File: pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2006.go [hunk 6]
```diff
@@ -93,9 +93,8 @@ func (r *Rule2006) Run(ctx context.Context) (rule.RuleResult, error) {
 					if strings.Contains(resource, "*") {
 						if accepted {
 							return rule.AcceptedCheckResult(msg, target)
-						} else {
-							return rule.FailedCheckResult("RBAC Role uses \"*\" in policy rule resources.", target)
 						}
+						return rule.FailedCheckResult("RBAC Role uses \"*\" in policy rule resources.", target)
```
Comment (AleksandarSavchev) [inline pkg/provider/managedk8s/ruleset/securityhardenedk8s/rules/2006.go:97]:
```suggestion
						return rule.FailedCheckResult("Role uses \"*\" in policy rule resources.", target)
```

**Review by AleksandarSavchev — APPROVED** [review]
/lgtm

**Review by dimityrmirchev — APPROVED** [review]
/lgtm

**Pull request merged**